### PR TITLE
More Debian package fixes (2)

### PR DESCRIPTION
**Pull request by marcosfrm**
Now I think Debian stuff is ok.

### CODE REVIEW

File: install/debian/postinst.in [hunk 1]
```diff
@@ -18,10 +19,10 @@ case "$1" in
     fi
 
     # Create directories
-    install -d -m 755 -o $NAME -g $NAME $CLIENT_HOME $CLIENT_LOGS
+    install -d -m 755 -o $NAME -g $NAME $CLIENT_CONF $CLIENT_HOME $CLIENT_LOGS
 
     # Create a symbolic link to the configuration file
-    ln -sf /etc/fahclient/config.xml $CLIENT_HOME/config.xml
+    ln -sf $CLIENT_HOME/config.xml $CLIENT_CONF/config.xml
```
Comment (jcoffland) [inline install/debian/postinst.in:25]: This looks backwards to me.

Comment (marcosfrm) [reply install/debian/postinst.in:25]:
```
NAME
       ln - make links between files

SYNOPSIS
       ln [OPTION]... [-T] TARGET LINK_NAME
[...]
```

It is backwards today...

```
# ls -l /var/lib/fah-client/
total 28
-rw-r--r-- 1 fah-client fah-client 28672 Sep 26 15:04 client.db
lrwxrwxrwx 1 root       root          25 Sep 26 15:04 config.xml -> /etc/fahclient/config.xml
```

**Comment by jcoffland** [comment]
I think we used to install the ``config.xml`` in ``/usr/lib/fah-client``.  Then the link was added so it could be found in ``/etc/fah-client/``.  But putting it in ``/etc/fah-client/`` in the first place makes more sense.  Currently, there is no ``config.xml`` by default.  However, we need to consider installs for the v7 client which may have a ``/var/lib/fah-client/config.xml``.  This would over write that.  The installer should probably not force the symlink.  It could overwrite the configuration.

**Comment by kbernhagen** [comment]
Installer used to make a link to /etc/fahclient/config.xml, which matched the location of v7 config. Note there is no hyphen in fahclient.

**Comment by kbernhagen** [comment]
My understanding was that the /etc location was only used so v7 config would be picked up easily.

**Comment by kbernhagen** [comment]
You can’t expect the v7 config to persist in the data directory, because such directory is probably deleted on v8 unstall/upgrade.

**Comment by kbernhagen** [comment]
The previous ln command was correct.

A copy might be reasonable if a non-link config doesn’t already exist in client home.

**Comment by kbernhagen** [comment]
I think this PR should be rejected.

**Comment by kbernhagen** [comment]
I agree the symlink should not be forced, for what it’s worth.

**Comment by marcosfrm** [comment]
OK, I missed that.

https://github.com/FoldingAtHome/fah-client-bastet/blob/f3687c091987bbff49c37a268140c0d9bef92c8b/install/debian/postrm#L11C1-L12

So `/etc/fahclient` should be removed here on purge?

**Comment by kbernhagen** [comment]
Dropping preinst sounds reasonable since the install does a systemctl restart on upgrade.

**Comment by kbernhagen** [comment]
> OK, I missed that.
> 
> https://github.com/FoldingAtHome/fah-client-bastet/blob/f3687c091987bbff49c37a268140c0d9bef92c8b/install/debian/postrm#L11C1-L12
> 
> So `/etc/fahclient` should be removed here on purge?

Maybe. Joseph would know.

**Comment by kbernhagen** [comment]
Maybe check the v7 purge. 

https://github.com/FoldingAtHome/fah-client/blob/master/debian/postrm

**Comment by kbernhagen** [comment]
Oops. You probably can’t see that.

**Comment by kbernhagen** [comment]
The v7 purge has 

rm -rf /var/lib/fahclient /etc/fahclient /etc/default/fahclient
        rm -rf /var/run/fahclient.pid

**Comment by marcosfrm** [comment]
I am thinking in v7->v8 update case. Users will have `/etc/fahclient/config.xml` and v8 purge will keep it there, since it is removing a non-existent `/etc/fah-client` directory.

**Comment by kbernhagen** [comment]
That sounds correct. But I’m not sure /etc/fah-client never exists.

**Comment by jcoffland** [comment]
I think the PR makes sense but it should avoid overwriting any pre-existing config.xml.

**Comment by kbernhagen** [comment]
The symlink needs to be to /etc/fahclientconfig.xml (no hyphen).

**Comment by kbernhagen** [comment]
And like you said, the ln args are reversed.

**Comment by kbernhagen** [comment]
Maybe need ||true if the symlink is not forced.

**Comment by marcosfrm** [comment]
I have dropped the second commit.

**Pull request merged**